src/org/eclipse/swt/widgets/Display.d

```diff
@@ -4182,61 +4186,69 @@ static dchar wcsToMbcs (char ch) {
-    return cbdata.display.windowProc( cast(GtkWidget*)handle, arg0, cast(int)cbdata.data );
+    return cbdata.display.windowProc( cast(GtkWidget*)handle, arg0, cast(ptrdiff_t)cbdata.data );
 }
-int /*long*/ windowProc (GtkWidget* handle, int /*long*/ arg0, int /*long*/ user_data) {
+int windowProc (GtkWidget* handle, ptrdiff_t arg0, ptrdiff_t user_data) {
     Widget widget = getWidget (handle);
     if (widget is null) return 0;
     return widget.windowProc (handle, arg0, user_data);
 }
 
-private static extern(C) int /*long*/ windowProcFunc4 (int /*long*/ handle, int /*long*/ arg0, int /*long*/ arg1, int /*long*/ user_data) {
+private static extern(C) int windowProcFunc4 (ptrdiff_t handle, ptrdiff_t arg0, ptrdiff_t arg1, ptrdiff_t user_data) {
     version(LOG) getDwtLogger().error( __FILE__, __LINE__,  "Display {}:", __LINE__ ).flush;
     CallbackData* cbdata = cast(CallbackData*)user_data;
-    return cbdata.display.windowProc( cast(GtkWidget*)handle, arg0, arg1, cast(int)cbdata.data );
+    return cbdata.display.windowProc( cast(GtkWidget*)handle, arg0, arg1, cast(ptrdiff_t)cbdata.data );
 }
-int /*long*/ windowProc (GtkWidget* handle, int /*long*/ arg0, int /*long*/ arg1, int /*long*/ user_data) {
+int windowProc (GtkWidget* handle, ptrdiff_t arg0, ptrdiff_t arg1, ptrdiff_t user_data) {
     Widget widget = getWidget (handle);
     if (widget is null) return 0;
     return widget.windowProc (handle, arg0, arg1, user_data);
 }
 
-private static extern(C) int /*long*/ windowProcFunc5 (int /*long*/ handle, int /*long*/ arg0, int /*long*/ arg1, int /*long*/ arg2, int /*long*/ user_data) {
+private static extern(C) int windowProcFunc5 (ptrdiff_t handle, ptrdiff_t arg0, ptrdiff_t arg1, ptrdiff_t arg2, ptrdiff_t user_data) {
     version(LOG) getDwtLogger().error( __FILE__, __LINE__,  "Display {}:", __LINE__ ).flush;
     CallbackData* cbdata = cast(CallbackData*)user_data;
-    return cbdata.display.windowProc( cast(GtkWidget*)handle, arg0, arg1, arg2, cast(int)cbdata.data );
+    return cbdata.display.windowProc( cast(GtkWidget*)handle, arg0, arg1, arg2, cast(ptrdiff_t)cbdata.data );
 }
-int /*long*/ windowProc (GtkWidget* handle, int /*long*/ arg0, int /*long*/ arg1, int /*long*/ arg2, int /*long*/ user_data) {
+int windowProc (GtkWidget* handle, ptrdiff_t arg0, ptrdiff_t arg1, ptrdiff_t arg2, ptrdiff_t user_data) {
     Widget widget = getWidget (handle);
     if (widget is null) return 0;
     return widget.windowProc (handle, arg0, arg1, arg2, user_data);
 }
 
+private static extern(C) int windowProcChangeValueFunc(GtkWidget* handle, int scroll, double value1, void* user_data) {
+    version(LOG) getDwtLogger().error( __FILE__, __LINE__,  "Display {}:", __LINE__ ).flush;
+    CallbackData* cbdata = cast(CallbackData*)user_data;
+    Widget widget = cbdata.display.getWidget (handle);
+    if (widget is null) return 0;
+    return widget.gtk_change_value(handle, scroll, value1, user_data);
+}
```

Comment:
Declaration of CHANGE_VALUE event doesn't match windowProcFunc5. If use windowProcFunc5, Occurs Segfault.

See also: https://developer.gnome.org/gtk3/stable/GtkRange.html#GtkRange-change-value